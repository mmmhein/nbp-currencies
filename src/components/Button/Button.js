import styled from 'styled-components';

import media from '../../styles/media';
import {
  colorTurquoise,
  colorWhite,
  colorYellow,
  fontWeightBold,
  borderWidthXThick,
  fontMedium,
  colorDarkGrey,
  fontSmall
} from '../../styles/designTokens';

const Button = styled.button`
  background: ${({ grey }) => (grey ? colorDarkGrey : colorTurquoise)};
  border: 0;
  color: ${colorWhite};
  display: block;
  font-size: ${fontMedium};
  font-weight: ${fontWeightBold};
  padding: 0.8rem;
  text-align: center;
  text-decoration: none;
  width: 100%;
  max-width: 15rem;

  &:not(:disabled) {
    cursor: pointer;
  }

  &:focus {
    outline: ${borderWidthXThick} solid ${colorYellow};
    outline-offset: 0;
  }

  ${media.phone`
    font-size: ${fontSmall}
    max-width: 9rem;
  `};
`;

export default Button;
