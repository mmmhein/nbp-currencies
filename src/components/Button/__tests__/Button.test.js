import React from 'react';
import { create } from 'react-test-renderer';

import Button from '../Button';

describe('COMPONENT - Button', () => {
  it('renders correctly', () => {
    const component = create(<Button />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly grey button', () => {
    const component = create(<Button grey />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
