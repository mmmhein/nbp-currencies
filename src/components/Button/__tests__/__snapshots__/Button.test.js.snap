// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Button renders correctly 1`] = `
<button
  className="sc-bdVaJa cyKlHL"
/>
`;

exports[`COMPONENT - Button renders correctly grey button 1`] = `
<button
  className="sc-bdVaJa ItcSe"
/>
`;
